main.go

```diff
@@ -66,12 +68,57 @@ func main() {
 		w.Write([]byte("OK\n"))
 	})
 
+	// Will be used with both http and https
+	// Inspired partially by https://gist.github.com/yowu/f7dc34bd4736a65ff28d
+	var schemaInferenceHandler = http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
+		// Do auth
+		// Pull JWT from authz header
+		// See auth.go:authorized()
+		// decodeJWT(that bearer token) -> AuthorizedClaims
+		claims, err := authorized_req(req)
+		if err != nil {
+			http.Error(writer, err.Error(), http.StatusUnauthorized)
+			return
+		}
+
+		collection_name := req.URL.Query().Get("collection_name")
```

Comment:
ubernit: Elsewhere in public APIs we've preferred the bare term `collection` when referring to a collection by name.